resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesSuite.scala

```diff
@@ -381,4 +451,34 @@ private[spark] class KubernetesSuite extends SparkFunSuite with BeforeAndAfter {
     val sparkMetricsService = getSparkMetricsService("spark-pi")
     expectationsForStaticAllocation(sparkMetricsService)
   }
+
+  test("Dynamic executor scaling basic test") {
+    createShuffleServiceDaemonSet()
+
+    val args = Array(
+      "--master", s"k8s://https://${Minikube.getMinikubeIp}:8443",
+      "--deploy-mode", "cluster",
+      "--kubernetes-namespace", NAMESPACE,
+      "--name", "spark-pi",
+      "--executor-memory", "512m",
+      "--jars", HELPER_JAR_FILE.getAbsolutePath,
+      "--class", GROUP_BY_MAIN_CLASS,
+      "--conf", "spark.ui.enabled=true",
+      "--conf", "spark.testing=false",
+      "--conf", "spark.dynamicAllocation.enabled=true",
+      "--conf", "spark.shuffle.service.enabled=true",
+      "--conf", "spark.dynamicAllocation.maxExecutors=3",
+      "--conf", s"spark.kubernetes.shuffle.namespace=${NAMESPACE}",
+      "--conf", "spark.kubernetes.shuffle.labels=app=spark-shuffle-service",
+      "--conf", s"${KUBERNETES_SUBMIT_CA_CERT_FILE.key}=${clientConfig.getCaCertFile}",
+      "--conf", s"${KUBERNETES_SUBMIT_CLIENT_KEY_FILE.key}=${clientConfig.getClientKeyFile}",
+      "--conf", s"${KUBERNETES_SUBMIT_CLIENT_CERT_FILE.key}=${clientConfig.getClientCertFile}",
+      "--conf", s"${EXECUTOR_DOCKER_IMAGE.key}=spark-executor:latest",
+      "--conf", s"${DRIVER_DOCKER_IMAGE.key}=spark-driver:latest",
+      "--conf", s"${WAIT_FOR_APP_COMPLETION.key}=false",
+      EXAMPLES_JAR_FILE.getAbsolutePath)
+    SparkSubmit.main(args)
```

Comment:
Use the `Client` class directly as using `spark-submit` manipulates system properties